### PR TITLE
Feat/input component

### PR DESCRIPTION
**Pull request by HellraiserCW**
Added reusable input component

### CODE REVIEW

File: src/app/shared/components/input/input.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div #socials class="epm-input__wrapper" [ngClass]="{ socials: icon }">
+  <input #input class="input" [name]="placeholder" [type]="type" (input)="onValueChange()" />
```
Comment (GennadiyShpak) [inline src/app/shared/components/input/input.component.html:2]: why do we need type?
and I don`t l like your solution about name

File: src/app/shared/components/input/input.component.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  <label [for]="placeholder" class="placeholder" [ngClass]="{ filled: input.value }">
+    {{ placeholder }}
+  </label>
+</div>
```
Comment (GennadiyShpak) [inline src/app/shared/components/input/input.component.html:6]: And, i think we can find better solution for this markup, let discuss on daily meeting

File: src/app/shared/components/input/input.component.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
+@import 'variables';
+
+:host {
+  position: relative;
+}
```
Comment (GennadiyShpak) [inline src/app/shared/components/input/input.component.scss:5]: why do we need it?

File: src/app/shared/components/input/input.component.scss [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    font-weight: $font-weight-medium;
+    color: $text-dark-grey;
+    border: 1px solid $color-light-grey;
+    border-radius: 8px;
+
+    &:focus {
+      outline: none;
+      border-color: $input-border-focused;
+      box-shadow: 0 0 0 1px $input-border-focused;
+    }
```
Comment (GennadiyShpak) [inline src/app/shared/components/input/input.component.scss:25]: Should we change focus styles?

File: src/app/shared/components/input/input.component.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  &.epm-input__wrapper {
+    .input {
+      padding-left: 48px;
+    }
+
+    .placeholder {
+      left: 48px;
+    }
+  }
+}
```
Comment (GennadiyShpak) [inline src/app/shared/components/input/input.component.scss:63]: you are repeat class names

File: src/styles/_variables.scss [hunk 6]
```diff
@@ -1,2 +1,14 @@
...
+$text-dark-grey: #25272d;
+$text-secondary: #66666f;
+
+$input-border-focused: #315cad;
+
+$font-size-label: 12px;
+$font-size-regular: 16px;
+
+$font-weight-regular: 400;
+$font-weight-medium: 500;
```
Comment (GennadiyShpak) [inline src/styles/_variables.scss:14]: why do wee need it?

File: src/app/shared/components/input/input.component.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<div #socials class="epm-input__wrapper" [ngClass]="{ socials: icon }">
```
[the remark this answers is not in the document]
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.html]: fixed

File: src/app/shared/components/input/input.component.html (hunk 1 above)
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.html]: fixed

File: src/app/shared/components/input/input.component.html (hunk 2 above)
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.html]: fixed

File: src/app/shared/components/input/input.component.scss (hunk 3 above)
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.scss]: fixed

File: src/app/shared/components/input/input.component.scss (hunk 4 above)
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.scss:45]: need to ask designers

File: src/app/shared/components/input/input.component.scss (hunk 5 above)
Comment (HellraiserCW) [reply src/app/shared/components/input/input.component.scss]: fixed

File: src/styles/_variables.scss (hunk 6 above)
Comment (HellraiserCW) [reply src/styles/_variables.scss]: fixed

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**